akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -542,6 +568,28 @@ object ClusterReceptionist {
     Props(new ClusterReceptionist(pubSubMediator, settings)).withDeploy(Deploy.local)
 
   /**
+   * Declares a super type for all events emitted by the `ClusterReceptionist`.
+   * Events are published to the Akka event stream (system.eventStream).
+   */
+  sealed abstract class ClusterReceptionistEvent
+
+  /**
+   * Emitted to the Akka event stream when a heartbeat has been
+   * received from a ClusterClient. Consumers of the event should expect
+   * to receive it periodically and at reasonably regular intervals.
+   *
+   * It is up to the receiver of this event to discern whether there are new
+   * cluster clients or whether the cluster clients have been removed since any
+   * prior `HeartbeatReceived` event.
+   *
+   * @param clusterClient The cluster client that sent the heartbeat
+   * @param clusterReceptionist The receptionist that emitted the event
+   */
+  final case class HeartbeatReceived(
+    clusterClient: ActorPath,
+    clusterReceptionist: ActorRef) extends ClusterReceptionistEvent
```

Comment:
I think publishing full state (default 3 contact points, once per minute) would be easier, but I'm fine with publishing deltas of contact points if you prefer that. Then you need to add the full state query also.